scanpy/preprocessing/_pca.py

```diff
@@ -151,19 +157,52 @@ def pca(
 
     logg.info(f'    with n_comps={n_comps}')
 
-    random_state = check_random_state(random_state)
-
     X = adata_comp.X
 
-    if chunked:
+    is_dask = isinstance(X, DaskArray)
+
+    random_state = check_random_state(random_state) if not is_dask else random_state
+
+    if is_dask and not chunked:
+        # we will only let the user choose the randomized solver otherwise
+        # we will use our default
+        if svd_solver != 'randomized' and svd_solver != 'auto':
+            logg.debug('Ignoring svd_solver')
+        if zero_center:
+            from dask_ml.decomposition import PCA as daskPCA
+
+            # will let dask handle if not randomized
+            algorithm = 'randomized' if svd_solver == 'randomized' else 'auto'
+            pca_ = daskPCA(
+                n_components=n_comps, svd_solver=algorithm, random_state=random_state
+            )
+            X_pca = pca_.fit_transform(X)
+        else:
+            from dask_ml.decomposition import TruncatedSVD as daskSVD
+
+            # daskSVD only supports randomized and tsqr
+            algorithm = 'randomized' if svd_solver == 'randomized' else 'tsqr'
+            pca_ = daskSVD(
+                n_components=n_comps, algorithm=algorithm, random_state=random_state
+            )
+            X_pca = pca_.fit_transform(X)
+    elif chunked:
         if not zero_center or random_state or svd_solver != 'arpack':
             logg.debug('Ignoring zero_center, random_state, svd_solver')
 
-        from sklearn.decomposition import IncrementalPCA
+        X_pca = None
+        pca_ = None
+        if is_dask:
+            import dask.array as da
+            from dask_ml.decomposition import IncrementalPCA as daskIncrementalPCA
 
-        X_pca = np.zeros((X.shape[0], n_comps), X.dtype)
+            pca_ = daskIncrementalPCA(n_components=n_comps)
+            X_pca = da.zeros((X.shape[0], n_comps), X.dtype)
+        else:
+            from sklearn.decomposition import IncrementalPCA
 
-        pca_ = IncrementalPCA(n_components=n_comps)
+            pca_ = IncrementalPCA(n_components=n_comps)
+            X_pca = np.zeros((X.shape[0], n_comps), X.dtype)
```

Comment:
```suggestion
        if is_dask:
            from dask.array import zeros
            from dask_ml.decomposition import IncrementalPCA
        else:
            from numpy import zeros
            from sklearn.decomposition import IncrementalPCA

        pca_ = IncrementalPCA(n_components=n_comps)
        X_pca = zeros((X.shape[0], n_comps), X.dtype)
```

do the same with the `pca_ =` and `X_pca =` lines above